module/VuFind/src/VuFind/Controller/FeedbackController.php

```diff
@@ -96,49 +98,42 @@ public function formAction()
             return $view;
         }
 
-        $fields = $form->mapRequestParamsToFieldValues($this->params()->fromPost());
-        $emailMessage = $this->getViewRenderer()->partial(
-            'Email/form.phtml',
-            compact('fields')
-        );
-
-        [$senderName, $senderEmail] = $this->getSender($form);
-
-        $replyToName = $params->fromPost(
-            'name',
-            $user ? trim($user->firstname . ' ' . $user->lastname) : null
-        );
-        $replyToEmail = $params->fromPost(
-            'email',
-            $user ? $user->email : null
-        );
-
-        $recipients = $form->getRecipient($params->fromPost());
-
-        $emailSubject = $form->getEmailSubject($params->fromPost());
-
-        $sendSuccess = true;
-        foreach ($recipients as $recipient) {
-            [$success, $errorMsg] = $this->sendEmail(
-                $recipient['name'],
-                $recipient['email'],
-                $senderName,
-                $senderEmail,
-                $replyToName,
-                $replyToEmail,
-                $emailSubject,
-                $emailMessage
+        $primaryHandler = $form->getPrimaryHandler();
+        $result = $primaryHandler->handle($form, $params, $user ?: null);
+        $success = $result['success'] ?? false;
+        if ($success) {
+            $view->setVariable(
+                'successMessage',
+                $result['successMessage'] ?? $form->getSubmitResponse()
             );
+            $view->setTemplate('feedback/response');
+        }
+        foreach ($result['errorMessages'] ?? [] as $error) {
+            $this->flashMessenger()->addErrorMessage($this->translate($error));
+        }
+        foreach ($result['errorMessagesDetailed'] ?? [] as $error) {
```

Comment:
Could logging of detailed errors be left to the handler? That would also eliminate the logging loop for the secondary handlers.